lisp/modules/rename_cues/rename_ui.py

```diff
@@ -0,0 +1,299 @@
+            self._cues_list.append({
+                'cue_name': cue.name,
+                'cue_preview': cue.name,
+                'selected': True,
+                'regex_groups': [],
+                'id': cue.id
+                })
+
+        self.update_preview_list()
+
+    def get_all_cues(self):
+        for cue in Application().cue_model.filter():
+            self._cues_list.append({
+                'cue_name': cue.name,
+                'cue_preview': cue.name,
+                'selected': True,
+                'regex_groups': [],
+                'id': cue.id
+            })
+
+        self.update_preview_list()
+
+    def update_preview_list(self):
+        if self.previewList.topLevelItemCount() == 0:
+            self.populate_preview_list()
+        else:
+            for i in range(self.previewList.topLevelItemCount()):
+                item = self.previewList.topLevelItem(i)
+                item.setText(0, self._cues_list[i]['cue_name'])
+                item.setText(1, self._cues_list[i]['cue_preview'])
+                if self._cues_list[i]['selected']:
+                    item.setCheckState(0, Qt.Checked)
+                else:
+                    item.setCheckState(0, Qt.Unchecked)
+
+    def populate_preview_list(self):
+        if self._cues_list != []:
+            for cue_to_rename in self._cues_list:
+                item = QTreeWidgetItem(self.previewList)
+                item.setText(0, cue_to_rename['cue_name'])
+                item.setText(1, cue_to_rename['cue_preview'])
+                if cue_to_rename['selected']:
+                    item.setCheckState(0, Qt.Checked)
+                else:
+                    item.setCheckState(0, Qt.Unchecked)
+
+    def onPreviewListItemPressed(self, item, event):
+        # Toggle checkbox on first column
+        # and change 'selected' in self._cues_list
+        i = self.previewList.indexOfTopLevelItem(item)
+        if item.checkState(0) == Qt.Checked:
+            item.setCheckState(0, Qt.Unchecked)
+            self._cues_list[i]['selected'] = False
+        else:
+            item.setCheckState(0, Qt.Checked)
+            self._cues_list[i]['selected'] = True
+
+    def onCapitalizeButtonClicked(self):
+        for cue in self._cues_list:
+            if cue['selected']:
+                cue['cue_preview'] = cue['cue_preview'].title()
+        self.update_preview_list()
+
+    def onLowerButtonClicked(self):
+        for cue in self._cues_list:
+            if cue['selected']:
+                cue['cue_preview'] = cue['cue_preview'].lower()
+        self.update_preview_list()
+
+    def onUpperButtonClicked(self):
+        for cue in self._cues_list:
+            if cue['selected']:
+                cue['cue_preview'] = cue['cue_preview'].upper()
+        self.update_preview_list()
+
+    def onRemoveNumButtonClicked(self):
+        regex = re.compile('^[^a-zA-Z]+(.+)')
+
+        def remove_numb(input):
```

Comment:
why defining this function? it seems an overkill, simply put this code in the for loop